src/lib/support/PersistentArray.h

```diff
@@ -0,0 +1,316 @@
+                return this->Save();
+            }
+        }
+        // Insert last
+        VerifyOrReturnError(this->mCount < this->mLimit, CHIP_ERROR_INVALID_LIST_LENGTH);
+        ReturnErrorOnFailure(this->Copy(mEntries[this->mCount++], value));
+        ReturnErrorOnFailure(this->Save());
+        OnEntryAdded(value);
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * @brief Returns the stored entry at a given index.
+     * @param index Zero-based position within the table.
+     * @param entry On success, contains the value matching the given index.
+     * @return CHIP_NO_ERROR on success,
+     *         CHIP_ERROR_NOT_FOUND if index is greater than the index of the last entry on the table.
+     */
+    CHIP_ERROR Get(size_t index, EntryType & value)
+    {
+        ReturnErrorOnFailure(this->Load());
+        VerifyOrReturnError(index < this->mLimit, CHIP_ERROR_NOT_FOUND);
+        VerifyOrReturnError(index < this->mCount, CHIP_ERROR_NOT_FOUND);
+        ReturnErrorOnFailure(this->Copy(value, mEntries[index]));
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * @brief Returns the value matching the given entry. The Compare() method
+     *        is used to identify the matching entry within the table.
+     * @param index Zero-based position within the table.
+     * @param entry Used as an input for comparison, and output for the rest of the values.
+     * @return CHIP_NO_ERROR on success,
+     *         CHIP_ERROR_NOT_FOUND if no entry matches the given input.
+     */
+    CHIP_ERROR Get(EntryType & value, bool do_create = false)
+    {
+        CHIP_ERROR err = this->Load();
+        VerifyOrReturnError(CHIP_ERROR_NOT_FOUND == err || CHIP_NO_ERROR == err, err);
+
+        // Check existing
+        for (size_t i = 0; i < this->mCount; ++i)
+        {
+            EntryType & e = mEntries[i];
+            if (this->Compare(value, e))
+            {
+                ReturnErrorOnFailure(this->Copy(value, e));
+                return CHIP_NO_ERROR;
+            }
+        }
+        // Insert last ?
+        VerifyOrReturnError(do_create, CHIP_ERROR_NOT_FOUND);
+        VerifyOrReturnError(this->mCount < this->mLimit, CHIP_ERROR_INVALID_LIST_LENGTH);
+        ReturnErrorOnFailure(this->Copy(mEntries[this->mCount++], value));
+        return this->Save();
+    }
+
+    /**
+     * @brief Removes the stored entry at a given index. Remaining entries are
+     *        shifted one position down.
+     * @param index Zero-based position within the table.
+     * @return CHIP_NO_ERROR on success,
+     *         CHIP_ERROR_NOT_FOUND if index is greater than the index of the last entry on the table.
+     */
+    CHIP_ERROR Remove(size_t index)
+    {
+        ReturnErrorOnFailure(this->Load());
+        VerifyOrReturnError(index < this->mLimit, CHIP_ERROR_NOT_FOUND);
+        VerifyOrReturnError(index < this->mCount, CHIP_ERROR_NOT_FOUND);
+        EntryType old = mEntries[index];
```

Comment:
So... this is using `operator=`, not Copy.  Should be documented why?  It's really not clear to me when Copy should be used or not used.

Maybe Copy needs a more descriptive name?

Reply:
This is an oversight. Copy should be use in all cases to ensure consistency.